### ISSUE TITLE
typeaheadWaitMs: Input string nullified during mid-flight async request

### ISSUE THREAD

**Issue by pageii** [valor-software/ngx-bootstrap#1724]
It's an edge case that breaks the directive.

--o----------------------------------------> User key-up
--------------------|--------------------|--> debounce (i.e. typeaheadWaitMs interval)
-------------X-----------------------------> User deletes entire input string (null).

Error message:

typeahead.directive.js:182 TypeError: Cannot read property 'subscribe' of null
    at Observable._subscribe (search.component.ts:60)
    at Observable._trySubscribe (Observable.js:57)
    at Observable.subscribe (Observable.js:45)
    at Object.subscribeToResult (subscribeToResult.js:22)
    at MergeMapSubscriber._innerSub (mergeMap.js:130)
    at MergeMapSubscriber._tryNext (mergeMap.js:127)
    at MergeMapSubscriber._next (mergeMap.js:110)
    at MergeMapSubscriber.Subscriber.next (Subscriber.js:89)
    at DebounceTimeSubscriber.debouncedNext (debounceTime.js:98)
    at AsyncAction.dispatchNext (debounceTime.js:114)
    at AsyncAction._execute (AsyncAction.js:111)
    at AsyncAction.execute (AsyncAction.js:86)
    at AsyncScheduler.flush (AsyncScheduler.js:36)
    at ZoneDelegate.webpackJsonp.1200.ZoneDelegate.invokeTask (zone.js:363)
    at Object.onInvokeTask (ng_zone.js:264)

In summary, if the user deletes the entire input (select all + delete) before the next debounce interval, typeahead breaks. The situation becomes more obvious as typeaheadWaitMs value increases.

**Comment by IlyaSurmay** [comment]
Please provide plunkr/stackblitz with reproduction if this is still relevant.
You can use one of starter templates:
Plunkr: https://plnkr.co/edit/0NipkZrnckZZROAcnjzB?p=preview
StackBlitz: https://stackblitz.com/edit/ngx-bootstrap?file=app%2Fapp.module.ts

**Comment by IlyaSurmay** [comment]
Closed as can't reproduce (tested with ngx-bootstrap@2.0.0-beta.8), will be reopened if we have sample to reproduce
